BrainPortal/public/javascripts/cbrain.js

```diff
@@ -790,6 +790,56 @@
       tool_tip.hide();
     });
 
+
+    // Adds copy to buffer button to all the element of class 'copy-text'.
+
+    if (navigator.clipboard) {
+      $(".copy-text").each(function () {
+        let $this = $(this);
+
+        // Keep line breaks
+        let text = $this.html()
```

Comment:
Why is this even necessary?

I think you're trying to extract the token? Why not extract it from a data-token attribute in HTML instead of the user facing HTML ? e.g.

```HTML
<div data-token="1836482537455723637374">
  <pretty>
    <something>
       1836482537455723637374
    </something>
   </pretty>
</div>
```

Basically, you fetch the token value from the attribute, not from the pretty HTML. That way you don't have to do any parsing, escaping etc.

Reply:
The goal of replacing `<br>` and `<p>` was to address multiline snippets, but at the moment it is not particularly needed and simple `.test()` worked fine in the previous version. The reliance on text rather than `data-` was to make the copy feature to easy to replicate without unnecessary duplication of code. Believe or not for a while I thought to add `data-` support as well, as an alternative option.